src/functional/ffront/foast_to_itir.py

```diff
@@ -320,6 +320,22 @@ def visit_BinOp(self, node: foast.BinOp, **kwargs) -> itir.FunCall:
             )
         )
 
+    def visit_TernaryExpr(self, node: foast.TernaryExpr, **kwargs) -> itir.FunCall:
+        true_value = to_value(node.true_expr)(self.visit(node.true_expr, **kwargs))
+        false_value = to_value(node.false_expr)(self.visit(node.false_expr, **kwargs))
+        cond_value = to_value(node.condition)(self.visit(node.condition, **kwargs))
+
+        if isinstance(node.true_expr.type, ct.ScalarType):
+            return im.call_("if_")(cond_value, true_value, false_value)
```

Comment:
```suggestion
```
Should be handled by `_lift_if_field` or am I missing something?

Reply:
not for this part, it works without (you can run the scan test to see it)